kafka_exporter.go

```diff
@@ -411,81 +446,111 @@ func (e *Exporter) collect(ch chan<- prometheus.Metric) {
 					topicPartitionLeader, prometheus.GaugeValue, float64(broker.ID()), topic, strconv.FormatInt(int64(partition), 10),
 				)
 			}
+			ch <- prometheus.MustNewConstMetric(
+				topicPartitions, prometheus.GaugeValue, float64(len(partitions)), topic,
+			)
 
-			currentOffset, err := e.client.GetOffset(topic, partition, sarama.OffsetNewest)
-			if err != nil {
-				glog.Errorf("Cannot get current offset of topic %s partition %d: %v", topic, partition, err)
-			} else {
-				e.mu.Lock()
-				offset[topic][partition] = currentOffset
-				e.mu.Unlock()
-				ch <- prometheus.MustNewConstMetric(
-					topicCurrentOffset, prometheus.GaugeValue, float64(currentOffset), topic, strconv.FormatInt(int64(partition), 10),
-				)
+			minInsyncReplicas := defaultMinInsyncReplicas
+			if _, ok := detail.ConfigEntries["min.insync.replicas"]; ok {
+				minInsyncReplicas, err = strconv.ParseFloat(*detail.ConfigEntries["min.insync.replicas"], 64)
 			}
-
-			oldestOffset, err := e.client.GetOffset(topic, partition, sarama.OffsetOldest)
 			if err != nil {
-				glog.Errorf("Cannot get oldest offset of topic %s partition %d: %v", topic, partition, err)
-			} else {
-				ch <- prometheus.MustNewConstMetric(
-					topicOldestOffset, prometheus.GaugeValue, float64(oldestOffset), topic, strconv.FormatInt(int64(partition), 10),
-				)
+				glog.Errorf("Could not parse minInsyncReplicas")
 			}
-
-			replicas, err := e.client.Replicas(topic, partition)
-			if err != nil {
-				glog.Errorf("Cannot get replicas of topic %s partition %d: %v", topic, partition, err)
-			} else {
+			if minInsyncReplicas != float64(-1) {
 				ch <- prometheus.MustNewConstMetric(
-					topicPartitionReplicas, prometheus.GaugeValue, float64(len(replicas)), topic, strconv.FormatInt(int64(partition), 10),
+					topicMinInsyncReplicas, prometheus.GaugeValue, float64(minInsyncReplicas), topic,
 				)
 			}
 
-			inSyncReplicas, err := e.client.InSyncReplicas(topic, partition)
-			if err != nil {
-				glog.Errorf("Cannot get in-sync replicas of topic %s partition %d: %v", topic, partition, err)
-			} else {
-				ch <- prometheus.MustNewConstMetric(
-					topicPartitionInSyncReplicas, prometheus.GaugeValue, float64(len(inSyncReplicas)), topic, strconv.FormatInt(int64(partition), 10),
-				)
-			}
+			e.mu.Lock()
```

Comment:
from this line, it also duplicated with line 437

Reply:
I'll work on this. It was merely a code rebase for now. Code quality has to be improved. I'll process your remarks. Thanks for reviewing !